Add docs about running tests with snowplow (#33159)
metabase/metabase: docs/developers-guide/e2e-tests.md

Please note the minus sign before the `@external` tag. For more details, consult [the official documentation](https://github.com/cypress-io/cypress-grep#filter-with-tags).

### Running tests with Snowplow involved

> Some tests depend on Snowplow and expect a running locally server. To run those tests, you need to:

Reviewer: Nit

```suggestion
Tests that depend on Snowplow expect a server running locally. To run them, you need to:
```
or

```suggestion
Tests that depend on Snowplow expect a locally running server. To run them, you need to:
```

or probably the most terse

```suggestion
Tests that depend on Snowplow expect a running server. To run them, you need to:
```
